tensorflow/core/common_runtime/dml/dml_tracing.cc

```diff
@@ -88,4 +178,24 @@ void DmlTracing::LogKernelCompute(const std::string& op_type,
                       TraceLoggingString(op_type.c_str(), "Type"),
                       TraceLoggingString(op_name.c_str(), "Name"));
   }
+}
+
+void DmlTracing::LogExecuteOperatorStart(
+    IDMLCompiledOperator* op, ID3D12GraphicsCommandList* command_list) {
+#if _WIN32
+  if (trace_level_ >= All) {
+    std::vector<char> eventName(100);
+    UINT data_size = (UINT)(eventName.size() * sizeof(char));
+    op->GetPrivateData(kPixEventNameId, &data_size, eventName.data());
```

Comment:
Does PIX just truncate if there's not enough space?

Reply:
DML truncates. See [IDMLObject::GetData](https://docs.microsoft.com/en-us/windows/win32/api/directml/nf-directml-idmlobject-getprivatedata). If we passed in data with 128 bytes and tried to read it out into a buffer with only 100 bytes we'd get the first 100 bytes.